### ISSUE TITLE
False positive for UPPER_CASE variable name

### ISSUE THREAD

**Issue by lesurp** [PyCQA/pylint#3132]
### Steps to reproduce

Code:
```python
import random

if __name__ == "__main__":
    number = random.randint(1, 10)
```

### Current behavior

```
************* Module ok
ok.py:1:0: C0114: Missing module docstring (missing-module-docstring)
ok.py:4:4: C0103: Constant name "number" doesn't conform to UPPER_CASE naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 3.33/10 (previous run: 3.33/10, +0.00)
```

The second error implies `number` is constant, which it is not.

### pylint --version output

```
pylint 2.4.0
astroid 2.3.0
Python 3.7.3 (default, May  9 2019, 15:28:41) 
[GCC 5.4.0 20160609]
```

**Comment by lesurp** [comment]
Note: it seems the bug is actually because pylint treats the `if __name__ == '__main__'` as module level code e.g.:

```python
import random


def compute(number):
    return 2 * number


if __name__ == "__main__":
    number = random.randint(1, 10)
```

produces:
```
************* Module ok
ok.py:1:0: C0114: Missing module docstring (missing-module-docstring)
ok.py:4:0: C0116: Missing function or method docstring (missing-function-docstring)
ok.py:4:12: W0621: Redefining name 'number' from outer scope (line 9) (redefined-outer-name)
ok.py:9:4: C0103: Constant name "number" doesn't conform to UPPER_CASE naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 2.00/10 (previous run: 2.00/10, +0.00)
```

The re-use of `number` should be fine here, but we have `Redefining name 'number' from outer scope (line 9) (redefined-outer-name)`

**Comment by PCManticore** [comment]
We probably should change the interpretation of "constant" as right now it's not entirely correct. There was another recent issue about the same thing and I agree that our interpretation of "constant" from PEP 8 was erroneous.
